tutorials/tour/_posts/2017-02-13-classes.md

```diff
@@ -9,46 +9,103 @@ categories: tour
+values, and variables, which are collectively called _members_.
+
+## Defining a class
+A minimal class definition is simply the keyword `class` and
+an identifier:
+```tut
+class User
+
+val user1 = new User
+```
+The keyword `new` is used to create an instance of the class. `User` has an implicit constructor which takes no arguments because no constructor was defined. However, you'll often want a constructor and class body. Here is an example class definition for a point:
 
 ```tut
 class Point(var x: Int, var y: Int) {
+
   def move(dx: Int, dy: Int): Unit = {
     x = x + dx
     y = y + dy
   }
+
   override def toString: String =
-    "(" + x + ", " + y + ")"
+    s"($x, $y)"
 }
+
+val point1 = new Point(2, 3)
+point1.x  // 2
+println(point1)  // prints (x, y)
 ```
 
-Classes in Scala are parameterized with constructor arguments. The code above defines two constructor arguments, `x` and `y`; they are both visible in the whole body of the class.
+This `Point` class has four members: the variables `x` and `y` and the methods `move` and
+`toString`. Unlike many other languages, the primary constructor is in the class signature `(var x: Int, var y: Int)`. The `move` method takes two integer arguments but does not return a value (the return type `Unit` corresponds to `void` in Java-like languages). `toString`, on the other hand, does not take any parameters but returns a `String` value. Since `toString` overrides the inherited `toString` method, it is tagged with the `override` keyword.
+
+## Constructors
+
+Constructors can have optional parameters by providing a default value like so:
 
-The class also includes two methods, `move` and `toString`. `move` takes two integer arguments but does not return a value (the return type `Unit` corresponds to `void` in Java-like languages). `toString`, on the other hand, does not take any parameters but returns a `String` value. Since `toString` overrides the pre-defined `toString` method, it is tagged with the `override` keyword.
+```tut
+class Point(var x: Int = 0, var y: Int = 0)
 
-Note that in Scala, it isn't necessary to say `return` in order to return a value. The value returned from a method is simply the last value in the method body. In the case of the `toString` method above, the expression after the equals sign is evaluated and returned to the caller.
+val origin = new Point  // x and y are both set to 0
+val point1 = new Point(1)
+println(point1.x)  // prints 1
 
-Classes are instantiated with the `new` primitive, as follows:
+```
 
+In this version of the `Point` class, `x` and `y` have the default value `0` so no arguments are required. However, because the constructor reads arguments left to right, if you just wanted to pass in a `y` value, you would need to name the parameter.
+```
+class Point(var x: Int = 0, var y: Int = 0)
+val point2 = new Point(y=2)
+println(point2.y)  // prints 2
+```
+
+This is also a good practice to enhance clarity.
+
+## Member scope
```

Comment:
I suggest renaming this section something like **Private Members and Getter/Setter Syntax**.